### PR TITLE
Display new per-stream and global state to users when viewing connection settings

### LINKED ISSUE #15017: When viewing per-stream state in the UI, it appears as if there is no state
When a connection is using per-stream state, it is not displayed in the UI.  The API does properly return the needed data.

```
GET https://cloud.airbyte.io/api/v1/state/get


{stateType: "stream", connectionId: "xxx",…}
connectionId: "xxx"
stateType: "stream"
streamState: [{streamDescriptor: {name: "users", namespace: "public"},…},…]
...}
```

### PR DESCRIPTION
**Pull request by evantahler**
Now with per-stream (and global) state, the `/workspaces/ID/connections/ID/settings` page needed to be updated to show one of the 3 possible state types, rather than only LEGACY state.

For a source using a per-stream  postgres source:

Before:
Screen Shot 2022-07-25 at 1 28 09 PM

After:
Screen Shot 2022-07-25 at 1 28 14 PM

Closes https://github.com/airbytehq/airbyte/issues/15017

### CODE REVIEW

**Comment by evantahler** [comment]
Note: to test this PR, I started by branching from https://github.com/airbytehq/airbyte/pull/15008 to have per-stream state enabled.  Then, after testing locally, I dropped all commits by mine

**Review by benmoriceau — APPROVED** [review]

**Comment by evantahler** [comment]
A note on the build process for @timroes: 

Running `npm build` or `./gradlew :airbyte-webapp:npm_run_build` locally said everything was OK, but the build system said I [had some lint errors](https://github.com/airbytehq/airbyte/runs/7508506636?check_suite_focus=true).  They were easy to fix, but why didn't my local run produce the same errors?  I tried with `NODE_ENV=production`, but that was a dead-end.

File: airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/components/StateBlock.tsx [hunk 1]
```diff
@@ -28,3 +28,16 @@ export const StateBlock: React.FC<StateBlockProps> = ({ connectionId }) => {
     </Card>
   );
 };
+
+function displayState(
```
Comment (tealjulia) [inline airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/components/StateBlock.tsx:32]: `calculateState`? `formatState`?  This doesn't have to do with displaying it really.

Comment (tealjulia) [inline airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/components/StateBlock.tsx:32]: I think this file is more readable if we move `displayState()` above `StateBlock()`.

**Review by teallarson — COMMENTED** [review]
Nice!

It sounds like a `globalState` might also have a `streamState` but the global state would supercede it.  Would a connection have `legacyState` and another kind of state ever?  

Other than that just nitpicks!

**Comment by evantahler** [comment]
> It sounds like a `globalState` might also have a `streamState` but the global state would supercede it. Would a connection have `legacyState` and another kind of state ever?

A connection might have /only/ legacy state, or we would store the state in one of the new formats (per stream or global)

**Review by teallarson — APPROVED** [review]
👍

**Comment by dizel852** [comment]
> A note on the build process for @timroes:
> 
> Running `npm build` or `./gradlew :airbyte-webapp:npm_run_build` locally said everything was OK, but the build system said I [had some lint errors](https://github.com/airbytehq/airbyte/runs/7508506636?check_suite_focus=true). They were easy to fix, but why didn't my local run produce the same errors? I tried with `NODE_ENV=production`, but that also said I had no errors. How could I have seen those format errors locally?

The PR is merged, but it seems like the question is still open.
So, the reason why you don't see any errors during the local build is because of the variable `CI=true` which is set by CI automatically. And all `warnings are treated as errors` (we can see the same message in logs). 
To simulate the same behavior locally we need just run `CI=true npm build`.
Hope this help in the future 🙏🏻

**Pull request merged**